web/src/routes/api/account/gmail/remove-email-labels/+server.ts

```diff
@@ -0,0 +1,201 @@
+ *
+ * @async
+ * @param {string} accessToken - The access token for the authenticated user.
+ * @param {string} threadId - The ID of the thread to check.
+ * @returns {Promise<boolean>} - A promise that resolves with a boolean indicating whether or not the thread ID corresponds to a Gmail thread.
+ * @throws {Error} - Throws an error if the API request fails or if the response is invalid.
+ */
+const isGmailThreadIdValid = async (accessToken: string, threadId: string): Promise<boolean> => {
+  // The Gmail thread endpoint, which gives the details of a specific Gmail thread
+  const threadEndpoint = `https://gmail.googleapis.com/gmail/v1/users/me/threads/${threadId}`;
+
+  try {
+    // Query the Gmail thread endpoint with the specified thread ID
+    const response = await fetch(threadEndpoint, {
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+        Accept: 'application/json',
+      },
+    });
+
+    // If the API returns a valid response, the thread ID is valid and corresponds to a Gmail thread
+    return response.ok;
+    
+  } catch (err) {
+    // If the API request fails or the response is invalid, throw an error
+    throw new Error(`Failed to retrieve information for thread ID: ${threadId}, corresponds to a Gmail thread: ${err.message}`);
+  }
+};
+
+/**
+ * Main handler function for this POST request route
+ *
+ * @param request - The incoming request object
+ * @param locals.getSession - function used to retrieve the current user's session.
+ * @param locals.supabase - the supabase client
+ * @returns {Response} - A response indicating whether or not the operation was successful.
+ * @throws Throws an error if the user is not authorized or if an unexpected error occurs.
+ */
+export const POST: RequestHandler = async ({ request, locals: { getSession, supabase } }) => {
+  
+  // Get the current user's session
+  const session = await getSession();
+  if (!session) throw error(401, 'unauthorized');
+
+  // Get the email and thread ID from the request body (if they exist)
+  let email = session.user.email;
+  let threadId = 'not present';
+  try {
+    const { user_email: reqEmail, email_thread_id: reqThreadId } = await request.json();
+    email = reqEmail || email;
+    threadId = reqThreadId || threadId;
+  } catch (err) {
+    // Do nothing
+  }
+
+  // Get a refreshed Google access token for the current user
+  let accessToken = '';
+  try {
+    accessToken = await getRefreshedGoogleAccessToken(supabase, email);
+  } catch (err) {
+    throw error(500, err instanceof Error ? err : 'unexpected error occurred');
+  }
+
+  // check the thread actually exists in Gmail before attempting the update
+  const isValidGmailThread = await isGmailThreadIdValid(accessToken, threadId)
+  if (!isValidGmailThread) {
+    return new Response({status: 200, statusText: "No corresponding Gmail thread"});
```

Comment:
This should be a `status: 400` or `status: 404`. Let's go with `404` because we can't find it